mail/postfix/src/opnsense/mvc/app/views/OPNsense/Postfix/senderbcc.volt

```diff
@@ -0,0 +1,111 @@
+this list of conditions and the following disclaimer.
+
+2.  Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+#}
+
+<script>
+
+    $( document ).ready(function() {
+        /*************************************************************************************************************
+         * link grid actions
+         *************************************************************************************************************/
+
+        $("#grid-senderbccs").UIBootgrid(
+            {   'search':'/api/postfix/senderbcc/searchSenderbcc',
+                'get':'/api/postfix/senderbcc/getSenderbcc/',
+                'set':'/api/postfix/senderbcc/setSenderbcc/',
+                'add':'/api/postfix/senderbcc/addSenderbcc/',
+                'del':'/api/postfix/senderbcc/delSenderbcc/',
+                'toggle':'/api/postfix/senderbcc/toggleSenderbcc/'
+            }
+        );
+
+        /*************************************************************************************************************
+         * Commands
+         *************************************************************************************************************/
+
+        /**
+         * Reconfigure
+         */
+        $("#reconfigureAct").click(function(){
+            $("#reconfigureAct_progress").addClass("fa fa-spinner fa-pulse");
+            ajaxCall("/api/postfix/service/reconfigure", {}, function(data,status) {
+                // when done, disable progress animation.
+                $("#reconfigureAct_progress").removeClass("fa fa-spinner fa-pulse");
+
+                if (status != "success" || data['status'] != 'ok') {
+                    BootstrapDialog.show({
+                        type: BootstrapDialog.TYPE_WARNING,
+                        title: "{{ lang._('Error reconfiguring Postfix') }}",
+                        message: data['status'],
+                        draggable: true
+                    });
+                } else {
+                    ajaxCall("/api/postfix/service/reconfigure", {});
+                }
+            });
+        });
+
+    });
```

Comment:
you may want to use include to deduplicate this:
https://docs.phalconphp.com/bs/3.3/volt#view-integration-include

Reply:
ok, before I change everything, you mean I put the reconfigure function into something like "apply.volt" and replace every reconfigure in these volts with `{% include 'OPNsense/Postfix/apply.volt' %}`

Reply:
Exactly if you want to deliver everything together or you can also use an external script.

Reply:
Interesting ... I should follow your coding more closely :)
I think I'm finished now, bit unsure about indent of the include, perhaps you have an advise?